Record tracking events for moving steps on the campaign creation and editing pages

eason9487 opened the pull request:
### Changes proposed in this Pull Request:

It's a follow-up PR of https://github.com/woocommerce/google-listings-and-ads/pull/2111#issuecomment-1777130207 and relates to #2104 

PR #2111 added missing tracking to the onboarding flow when moving steps, but the campaign creation and editing pages with the same `<Stepper>` also missed the tracking events.

This RP:

- Add utility functions to record events for moving steps on a page that has `<Stepper>`.
- Record tracking events for moving steps on the campaign creation and editing pages.

### Screenshots:

#### 📷 Campaign creation

![image](https://github.com/woocommerce/google-listings-and-ads/assets/17420811/82af23da-6df1-42e6-a4ea-ae97236627fc)

#### 📷 Campaign editing

![image](https://github.com/woocommerce/google-listings-and-ads/assets/17420811/59952e26-83a4-4bae-af05-01475c69cbf8)

### Detailed test instructions:

1. Enable tracking debug tool
   - Open the Console tab of the browser DevTool.
   - Change the filter level to "All levels".
   - Run `localStorage.setItem( 'debug', 'wc-admin:*' )` and refresh the page to make it work.
   - It should be able to view tracking via the Console tab. Ref:
      ![1](https://github.com/woocommerce/google-listings-and-ads/assets/17420811/698f85d8-e890-4ab6-939b-738acee6098f)
2. Go to the campaign creating page.
3. Click on the "Continue" button at the bottom to move to step 2.
4. Check the tracking event via the Console tab.
5. Click on the Stepper to move to step 1.
6. Check the tracking event via the Console tab.
7. Go to the campaign editing page.
8. Repeat steps 3-6 of this test instruction.

### Changelog entry

> Add - Record tracking events for moving steps on the campaign creation and editing pages.

eason9487 committed:
Add utility functions to record events for moving steps on a page that has `Stepper`.

eason9487 committed:
Add a new constant to map the step keys `CAMPAIGN_STEP` to step numbers in string type.

eason9487 committed:
Record events for moving steps on the campaign creation and editing pages.

eason9487 committed:
Update src/Tracking/README.md

jorgemd24 commented on js/src/pages/edit-paid-ads-campaign/index.js:
```diff
@@ -38,6 +48,9 @@ function getCurrentStep() {
 /**
  * Renders the campaign editing page.
+ *
+ * @fires gla_paid_campaign_step with `{ conext: 'edit-ads', triggered_by: 'step1-continue-button', action: 'go-to-step2' }`.
```
Typo: conext -> context

eason9487 replied:
Thanks for catching the typos. Fixed in https://github.com/woocommerce/google-listings-and-ads/pull/2140/commits/11f378c8fb2f3491251fbe35d439b2eb73261937.

jorgemd24 commented on js/src/pages/create-paid-ads-campaign/index.js:
```diff
@@ -23,13 +23,25 @@ import AdsCampaign from '.~/components/paid-ads/ads-campaign';
+ *
+ * @fires gla_paid_campaign_step with `{ conext: 'create-ads', triggered_by: 'step1-continue-button', action: 'go-to-step2' }`.
+ * @fires gla_paid_campaign_step with `{ conext: 'create-ads', triggered_by: 'stepper-step1-button', action: 'go-to-step1' }`.
  */
```
Typo: conext -> context

eason9487 replied:
Thanks for catching the typos. Fixed in https://github.com/woocommerce/google-listings-and-ads/pull/2140/commits/11f378c8fb2f3491251fbe35d439b2eb73261937.

jorgemd24 commented on js/src/utils/recordEvent.js:
```diff
@@ -119,6 +119,46 @@ export const recordTablePageEvent = ( context, page, direction ) => {
+ * @param {string} to The next step to go to, e.g. '2'.
+ * @param {string} [context] Indicates where this event happened.
+ */
+export function recordStepperChangeEvent( eventName, to, context ) {
+	recordEvent( eventName, {
+		triggered_by: `stepper-step${ to }-button`,
+		action: `go-to-step${ to }`,
+		context,
+	} );
+}
```
Wouldn't it be more valuable to also capture the starting step where the user was? Otherwise, the event would only indicate something simple, like "Button X was pressed and moved to Step X."

![image](https://github.com/woocommerce/google-listings-and-ads/assets/2488994/025d99f8-af4e-47ef-88f3-416db6ac6545)

eason9487 replied:
I guess the need to record the starting step would depend on the users of tracking data. This newly added event tracking was meant to align with the event properties of the other two:
- [gla_setup_mc](https://github.com/woocommerce/google-listings-and-ads/blob/11f378c8fb2f3491251fbe35d439b2eb73261937/src/Tracking/README.md#gla_setup_mc)
- [gla_setup_ads](https://github.com/woocommerce/google-listings-and-ads/blob/11f378c8fb2f3491251fbe35d439b2eb73261937/src/Tracking/README.md#gla_setup_ads)

If necessary, it would be a bit better to adjust the three event tracking together in a separate PR.

jorgemd24 replied:
Thanks for the explanation. From my perspective, having knowledge of the initial step would provide a bit more context, but I'm not someone who regularly uses tracking data.

jorgemd24 reviewed (commented):
Thanks, @eason9487, for implementing this tracking event – it works 👍 . I've left a comment on the `recordStepperChangeEvent` function as I'm uncertain whether it adds enough context to the tracking data.

eason9487 committed:
Fix typos.

Address:
- https://github.com/woocommerce/google-listings-and-ads/pull/2140#discussion_r1374353610
- https://github.com/woocommerce/google-listings-and-ads/pull/2140#discussion_r1374354553

eason9487 commented:
Hi @jorgemd24, thanks for the review.

This PR is ready for the second round of code reviews. Could you take another look?

jorgemd24 reviewed (approved):
Thanks @eason9487 for the clarifications and fixing the typos! LGTM 👍
